File: programmers/level2/구명보트.py
import collections
def solution(people, limit):
    people = collections.deque(sorted(people))
    weight = 0
    boat = 0
    while people:
        while people and weight+people[-1]<=limit:
            weight += people.pop()
        while people and weight+people[0]<=limit:
            weight += people.popleft()
        boat += 1
        weight = 0
    return boat

File: programmers/level2/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people);
        int leftIdx = 0;
        int rightIdx = people.length-1;
        int nowWeight = 0;
        while(leftIdx <= rightIdx) {
            while(leftIdx <= rightIdx && nowWeight+people[rightIdx] <= limit) {
                nowWeight += people[rightIdx--];
            }
            while(leftIdx <= rightIdx && nowWeight+people[leftIdx] <= limit) {
                nowWeight += people[leftIdx++];
            }
            answer++;
            nowWeight = 0;
        }
        return answer;
    }
}